flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala

```diff
@@ -225,12 +231,29 @@ abstract class StreamTableEnvironment(
     // decorrelate
     val decorPlan = RelDecorrelator.decorrelateQuery(relNode)
 
+    val prePlanner = createHepPlanner
```

Comment:
I'll merge PR #3101 later today which adds a normalization phase before optimization by adding a HepPlanner.
Could you integrate you changes with #3101?